sdk/schema/executable.go

```diff
@@ -75,9 +104,45 @@ func (e Executable) Validate() (bool, ValidationReport) {
 		Severity:    ValidationSeverityError,
 	})
 
+	report.AddCheck(ValidationCheck{
+		Description: "Credential Usages have either a credential reference or a selection form correctly defined",
+		Assertion:   CredentialUsagesAreProperlyDefined(e),
+		Severity:    ValidationSeverityError,
+	})
+
+	report.AddCheck(ValidationCheck{
+		Description: "Credential Usages are uniquely identifiable inside an executable",
+		Assertion:   NoDuplicateCredentialUsages(e),
+		Severity:    ValidationSeverityError,
+	})
+
 	return report.IsValid(), report
 }
 
 func (e Executable) Command() string {
 	return strings.Join(e.Runs, " ")
 }
+
+func (c CredentialUsage) GetCredentialReference() *CredentialReference {
+	if c.Name != "" || c.Plugin != "" || c.Provisioner != nil {
```

Comment:
I think we end up with an easier-to-debug situation if we only look at `c.Name` here. We always expect that to be set, right?

Reply:
Yes, but we still need a nice way of telling the user "Oh-oh you have specified a `plugin` or a `provisioner` but those will be ignored if you haven't specified a `Name`  or if you otherwise specified a `SelectFrom`". It helps them setting the right expectations on what they are defining.

Reply:
In other words, if a user specifies `Provisioner` or `Plugin` there might be intent there, and we should not ignore that intent, but rather catch it and let the user know that something is off.

Reply:
I made this function return an error in case a provisioner or plugin is defined without a credential name